### DISCUSSION TITLE
[Feature Request] Trade Module

### DISCUSSION THREAD

**Discussion by Stiphr** [quer/the-steam-awards#69] [Ideas]
Trade Inventory to specific account
Something like this https://github.com/Stiphr/Steam-Bulk-Trade-Bot

**Comment by quer** [comment]
Hey, i did look into this at some point. But i cant remeber why i did not add it.
but have just done a quick google search. And i cant find any trade npm module. That work whit the npm `steam`. 
all and like the one you send , is build around npm `steamcommunity`. 

Eather i have to change from using the npm `steam`, to  npm `steamcommunity`. or find one that is build on the npm `steam`

**Comment by Stiphr** [comment]
npm `steam` package provides a lower-level interface to interact with the Steam network compared to npm `steamcommunity` also, npm`steam` is outdated. I think you should consider build using `steamcommunity`

**Comment by HenkerX64** [comment]
steamcommunity have a rate limit for logins upto 30 seconds between logins, maybe `steam-user` then.

But to use steamcommunity it is enough to create object and clone cookies from websession
`const _community = new SteamCommunity();`
`_community.setCookies(cookies);`
try it out

---
to make a trade with second step, you can use `steam-tradeoffer-manager` package like this:
```
                const TradeOfferManager = require('steam-tradeoffer-manager');
		const manager = new TradeOfferManager({
			'language': 'en',
			'pollInterval': 7000,
			'community': _community,
		});
               manager.createOffer(steamId, token)
```

**Comment by quer** [comment]
You are right that npm `steam` is a lower-level interface, That is what i like about it, that there is not a lot off over head stuff, that is called. And that i can build the request part, to fit, and work as i want it to.

As in `SteamCommunity`, there is a components, that will use it own request module, insted of the once that i have build the hole project around. But the bigges problem is that, `SteamCommunity`, doent connect to the steam socket, So all modules that need to play a game, cant. as it more or less just get the cookies. and then do request whit them. 

Then we will have to be both. That is too much, to add lib like `SteamCommunity`, just to do Trading. I did see that you can just paste in Cookies into `SteamCommunity`, and then it will use that. But i am at the state where i will better do the rewite the trading module, to fit what i have. then adding more lib, to the project. to just get 1 module to work. 

i did look into `steam-user`, but this line here. made be stop:
https://github.com/DoctorMcKay/node-steam-user/blob/a8c0dd16304fea728851f840a58c14c04595c1ef/components/09-logon.js#L650
As it force the Console to enter a new Steam Guard, But that is a big No go. As i have build this up, to run multi account at once. Then it doent work, that it is running eks 10 at once, and 1 or maby more, will make the project stop, and wait. and the message it self, doent say what account it it using. 


So even that the npm `steam` is outdated. it is from what i see the best fit, to what this project provide. 

and currenly i doent see fit, to use time on make the npm `steam-tradeoffer-manager` fit into the project/module. as there is so many exampel on how to empty a account to a other account. So once you have to do that. Then just run that.

**Reply by HenkerX64** [reply]
to handle 'steamGuard' you need to register a listener like this:

```
steamUser.on('steamGuard', (domain, callback, lastCodeWrong) => {
// handle guard 
});
```
Then this code block `if (this.listenerCount('steamGuard') == 0) {` will get false and readline skipped

**Reply by HenkerX64** [reply]
About own request module and steamcommunity, it is possible to use yours one, just set it in steamcommunity options.

About `steam-tradeoffer-manager` and `steamcommunity`, I think it can be implemented as peerDependencies, if the user installed it, then the module can be used, if not then not. But I am not sure if it break the scope of the project, because it should solve the awards cases and trading is different scope.

**Reply by quer** [reply]
Some how i did not noted the `if (this.listenerCount('steamGuard') == 0) {` part. i have been looking at that npm module, to switch over. But that part stopped me.. I will look into it, at some point. 

i will have to look up peerDependencies, as i am not sure how it will work. 

I can maby do so the module only work, if you have manuel install some npm packages. So it just check before, it used. if all Dependencies have been installed. and then it can be used. That is not a bad ideer. As i did have the same problem whit csgo. as i made a module for somefing in the gameCoordinator. then the module will only work, if one have installed the npm package that is needed
